git-lfs.go

```diff
@@ -1,11 +1,37 @@
 package main
 
 import (
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
 	"github.com/github/git-lfs/commands"
 	"github.com/github/git-lfs/lfs"
 )
 
 func main() {
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, os.Kill)
+
+	var once sync.Once
+
+	go func() {
+		for {
+			sig := <-c
+			once.Do(lfs.ClearTempObjects)
+			fmt.Fprintf(os.Stderr, "\nExiting because of %q signal.\n", sig)
+
+			var exitCode int
```

Comment:
Being super pedantic signal exits are supposed to be 129+ (signal value is never 0) so to avoid that if the cast below fails, better to initialise exitCode to a non-zero value.